Tasks/NuGetCommand/task.json

```diff
@@ -0,0 +1,447 @@
+            "required": true,
+            "groupName": "packOptions",
+            "options": {
+                "off": "Off",
+                "byPrereleaseNumber": "Use the date and time",
+                "byEnvVar": "Use an environment variable",
+                "byBuildNumber": "Use the build number"
+            }
+        },
+        {
+            "name": "includeReferencedProjects",
+            "type": "boolean",
+            "label": "Include referenced projects",
+            "defaultValue": "false",
+            "helpMarkDown": "Include referenced projects either as dependencies or as part of the package. Cannot be used with automatic package versioning. If a referenced project has a corresponding nuspec file that has the same name as the project, then that referenced project is added as a dependency. Otherwise, the referenced project is added as part of the package. [Learn more](https://docs.nuget.org/consume/command-line-reference#pack-command) at NuGet.org.",
+            "groupName": "packOptions",
+            "visibleRule": "versioningScheme = off"
+        },
+        {
+            "name": "versionEnvVar",
+            "type": "string",
+            "label": "Environment variable",
+            "defaultValue": "",
+            "helpMarkDown": "Enter the variable name without $, $env, or %.",
+            "required": true,
+            "groupName": "packOptions",
+            "visibleRule": "versioningScheme = byEnvVar"
+        },
+        {
+            "name": "requestedMajorVersion",
+            "type": "string",
+            "label": "Major",
+            "defaultValue": "1",
+            "helpMarkDown": "The 'X' in version [X.Y.Z](http://semver.org/spec/v1.0.0.html)",
+            "required": true,
+            "groupName": "packOptions",
+            "visibleRule": "versioningScheme = byPrereleaseNumber"
+        },
+        {
+            "name": "requestedMinorVersion",
+            "type": "string",
+            "label": "Minor",
+            "defaultValue": "0",
+            "helpMarkDown": "The 'Y' in version [X.Y.Z](http://semver.org/spec/v1.0.0.html)",
+            "required": true,
+            "groupName": "packOptions",
+            "visibleRule": "versioningScheme = byPrereleaseNumber"
+        },
+        {
+            "name": "requestedPatchVersion",
+            "type": "string",
+            "label": "Patch",
+            "defaultValue": "0",
+            "helpMarkDown": "The 'Z' in version [X.Y.Z](http://semver.org/spec/v1.0.0.html)",
+            "required": true,
+            "groupName": "packOptions",
+            "visibleRule": "versioningScheme = byPrereleaseNumber"
+        },
+        {
+            "name": "buildProperties",
+            "type": "string",
+            "label": "Additional build properties",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Semicolon delimited list of properties used to build the package.",
```

Comment:
Might as well go into more detail here - a list of token=value pairs, separated by semicolons, where each occurrence of $token$ in the .nuspec file will be replaced with the given value